fedimint-server/src/config.rs

```diff
@@ -35,32 +35,47 @@ use crate::{ReconnectPeerConnections, TlsTcpConnector};
 const DEFAULT_MAX_CLIENT_CONNECTIONS: u32 = 1000;
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
+/// All the serializable configuration for the fedimint server
 pub struct ServerConfig {
-    pub federation_name: String,
-    pub identity: PeerId,
-    pub hbbft_bind_addr: String,
-    pub api_bind_addr: String,
-    #[serde(with = "serde_tls_cert")]
-    pub tls_cert: rustls::Certificate,
+    /// Contains all configuration that needs to be the same for every server
+    pub consensus: ServerConfigConsensus,
+    /// Contains all configuration that is locally configurable and not secret
+    pub local: ServerConfigLocal,
+    /// Contains all configuration that is secret such as private key material
+    pub private: ServerConfigPrivate,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ServerConfigPrivate {
     #[serde(with = "serde_tls_key")]
     pub tls_key: rustls::PrivateKey,
-
-    pub peers: BTreeMap<PeerId, Peer>,
     #[serde(with = "serde_binary_human_readable")]
     pub hbbft_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
     #[serde(with = "serde_binary_human_readable")]
-    pub hbbft_pk_set: hbbft::crypto::PublicKeySet,
+    pub epoch_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
+    pub modules: BTreeMap<String, serde_json::Value>,
+}
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ServerConfigConsensus {
+    pub federation_name: String,
+    pub peers: BTreeMap<PeerId, Peer>,
     #[serde(with = "serde_binary_human_readable")]
-    pub epoch_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
+    pub hbbft_pk_set: hbbft::crypto::PublicKeySet,
     #[serde(with = "serde_binary_human_readable")]
     pub epoch_pk_set: hbbft::crypto::PublicKeySet,
+    pub modules: BTreeMap<String, serde_json::Value>,
+}
```

Comment:
I wonder if we should just share the entire consensus config with the client and hash it to generate unique federation IDs. Related to #754. Alternatively we'd use a subset of this data (e.g. no HBBFT key, only the combined epoch pub key instead of the set) to generate a stable federation id.

Reply:
I think the hash of the entire consensus config would make for the safest kind of id.  The configs are small enough to pass around.

Reply:
The problem is the client doesn't have the pub key set but only the aggregated pub key, so could not calculate that hash right now. So either we change the client config to be a superset of `ServerConfigConsensus` or we need to hash a subset of that if we want to also allow clients to check they are talking to the right federation.

Reply:
Having the client config be a superset sounds like a good idea in terms of avoiding config duplication, but I'd have to look into whether that makes sense from a client standpoint.

Either way, we can expose an endpoint to retrieve the consensus config (or hash of it) for both server and client validation.